src/javascript/html-template-literal-scanner.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as babel from 'babel-types';
+
+import {ScannedInlineDocument} from '../model/model';
+
+import {getIdentifierName} from './ast-value';
+import {Visitor} from './estree-visitor';
+import {JavaScriptDocument} from './javascript-document';
+import {JavaScriptScanner} from './javascript-scanner';
+
+export class InlineHtmlDocumentScanner implements JavaScriptScanner {
+  async scan(
+      document: JavaScriptDocument,
+      visit: (visitor: Visitor) => Promise<void>) {
+    const features: ScannedInlineDocument[] = [];
+
+    const myVisitor: Visitor = {
+      enterTaggedTemplateExpression(node) {
+        const inlineDocument = getInlineDocument(node, document);
+        if (inlineDocument !== undefined) {
+          features.push(inlineDocument);
+        }
+      }
+    };
+
+    await visit(myVisitor);
+
+    return {features};
+  }
+}
+
+export function getInlineDocument(
+    node: babel.TaggedTemplateExpression,
+    parsedDocument: JavaScriptDocument): ScannedInlineDocument|undefined {
+  const tagName = getIdentifierName(node.tag);
+  if (tagName === undefined || !/(^|\.)html$/.test(tagName)) {
+    return;
+  }
+
+  const sourceRangeForLiteral = parsedDocument.sourceRangeForNode(node.quasi);
+  if (sourceRangeForLiteral === undefined) {
+    return;
+  }
+  const sourceRangeForContents = {
```

Comment:
If SourceRange were a class it could have some nice transformation methods...

Reply:
+1. Punting for now though.